packages/nuxi/src/commands/dev.ts

```diff
@@ -89,6 +90,14 @@ export default defineNuxtCommand({
         }
 
         currentNuxt = await loadNuxt({ rootDir, dev: true, ready: false })
+        // Hard restart
+        if (process.env.NUXI_CLI_WRAPPER) {
+          currentNuxt.hooks.hook('restart', (options) => {
+            if (options?.hard) {
+              process.exit(EXIT_CODE_RESTART)
+            }
```

Comment:
Refactor: We could merge it with the `restart` hook logic. Use hook once, and only hard reset when `opts.hard` ++ `NUXI_CLI_WRAPPER`